railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -184,6 +187,10 @@ def initialize(*args)
         if !options[:skip_active_record] && !DATABASES.include?(options[:database])
           raise Error, "Invalid value for --database option. Supported for preconfiguration are: #{DATABASES.join(", ")}."
         end
+
+        # Force sprockets to be skipped when generating http only app.
+        # Can't modify options hash as it's frozen by default.
+        self.options = options.merge(skip_sprockets: true, skip_javascript: true).freeze if options[:api]
```

Comment:
@rafaelfranca to do what you're mentioning we would need to change things like ...

1) https://github.com/rails/rails/blob/master/railties/lib/rails/generators/app_base.rb#L174
2) https://github.com/rails/rails/blob/master/railties/lib/rails/generators/app_base.rb#L170

The most trivial change ends being like https://gist.github.com/spastorino/7989a586151e8da13351, not just adding one method.

I don't think the diff is much better than what we already have but we should reconsider following a better approach in the future. Something around the lines of having an options class instead of a hash and defining methods in it for each possibility.